pkg/recipes/terraform/install.go

```diff
@@ -78,5 +81,40 @@ func Install(ctx context.Context, installer *install.Installer, tfDir string) (s
 
 	logger.Info(fmt.Sprintf("Terraform latest version installed to: %q", execPath))
 
-	return execPath, nil
+	// Create a new instance of tfexec.Terraform with current Terraform installation path
+	tf, err := NewTerraform(ctx, tfDir, execPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Verify Terraform installation is complete before proceeding
+	for attempt := 0; attempt <= installRetryCount; attempt++ {
+		_, _, err = tf.Version(ctx, false)
+		if err != nil {
+			if attempt < installRetryCount {
```

Comment:
Why do we need to do it this way?

Reply:
We want to return an error if we've hit the max retries. The if statement is there to check if we've the max retires and we still have an error. I can't respond to the last comment directly for some reason but we need the `if case` for this reason